File: Ejercicios/2. Starbuzz (Beverages)/Net/After/Starbuzz/Whip.cs
namespace Starbuzz
{
	public class Whip : CondimentDecorator
	{
		Beverage beverage;
		
		public Whip(Beverage beverage)
		{
			this.beverage = beverage;
		}

		public override double Cost()
		{
			return .10 + this.beverage.Cost();
		}
	}
}

File: Ejercicios/2. Starbuzz (Beverages)/Java/After/src/main/starbuzz/Whip.java
package starbuzz;
 
public class Whip extends CondimentDecorator {
	Beverage beverage;
 
	public Whip(Beverage beverage) {
		this.beverage = beverage;
	}

	public double cost() {
		return .10 + beverage.cost();
	}
}
